src/TorchSharp/Tensor/Tensor.cs

```diff
@@ -27,11 +27,22 @@ public partial class Tensor : IDisposable
             static long _totalCount = 0;
             static long _peakCount = 0;
 
+
+            public static List<Action<Tensor>> OnTensorCreateCallback { get; } =
+                new List<Action<Tensor>>();
+
+            public static List<Action<Tensor>> OnTensorDisposeCallback { get; } =
+                new List<Action<Tensor>>();
```

Comment:
These should not be public, or they should be events.

Also, name for the events should be `OnTensorCreated` and `BeforeTensorDisposed` correspondingly.

Reply:
The lists shouldn't be publicly accessible -- that would mean that you can remove hooks, or manipulate things in weird ways. There should be functions to register hooks.

Reply:
Changed to this;

            internal delegate void TensorEvent(Tensor tensor);
            internal static event TensorEvent? OnTensorCreated;
            internal static event TensorEvent? BeforeTensorDisposed;

            internal Tensor(IntPtr handle)
            {
                this.handle = handle;
                System.Threading.Interlocked.Increment(ref _totalCount);
                _peakCount = Math.Max(_totalCount, _peakCount);
                OnTensorCreated?.Invoke(this);
            }

Reply:
I'm no longer using these events, every tensor knows what scope it belongs to and communicates directly with that scope as it's created and disposed.

However, there may be a need for the events for other reasons, should I remove them or leave them in?

Reply:
I'm removing them for now, if they're needed/wanted, they should be added in a separate PR. 

I for one would like to have them, though.